qcodes/instrument/channel.py

```diff
@@ -32,14 +32,20 @@ def __init__(self,
                  parent: Union[Instrument, 'InstrumentChannel'],
                  name: str,
                  **kwargs) -> None:
-        # Initialize base classes of Instrument. We will overwrite what we
-        # want to do in the Instrument initializer
+        # need to specify parent before `super().__init__` so that the right
+        # `full_name` is available in that scope. `full_name` is used for
+        # registering the filter for the log messages. It is composed by
+        # iteratively concatenating the full names of the parent instruments
+        # scope of `Base`
+        self._parent = parent
         super().__init__(name=name, **kwargs)
-
+        # Naming insanity:
+        # (see https://github.com/QCoDeS/Qcodes/issues/1140 for a nice table)
+        # this has been a confusion about names. don't use name but
```

Comment:
"don't use" for what?

Reply:
for anything ;-)

Reply:
but then the next line of code uses `self.name` and `parent.name`.

i'm confused. the discussion in the link tells to use short_name everywhere (so far as i understood) because it is consistent among insrtument, parameter, channel.

Reply:
yes the `self.name` is an assignment. You cannot remove it because you would break compatibility. The use of `parent.name` could be indeed improved. That part has been there before Jens introduced the concept of `short_name`. Feel free to go through the code base and remove all those.